src/traversals/dfs.jl

```diff
@@ -5,12 +5,37 @@
     is_cyclic(g)
 
 Return `true` if graph `g` contains a cycle.
-
+  
 ### Implementation Notes
-Uses DFS.
+The algorithm uses a DFS.
 """
 function is_cyclic end
-@traitfn is_cyclic(g::::(!IsDirected)) = ne(g) > 0
+@enum Vertex_state unvisited visited
+@traitfn function is_cyclic(g::AG::(!IsDirected)) where {T, AG<:AbstractGraph{T}}
+    vertex_state = Vector{Vertex_state}(undef, nv(g)) 
+    fill!(vertex_state,unvisited)
```

Comment:
I agree with you but, in a second moment, I would like to improve the part of the code for the directed graphs. In that case there are present tree states. So I wouldn't change this in order to have the same style. What do you think?

Reply:
I think I see your point, but I believe having the precise same style in the two functions is not crucial though, because the involved algorithms are not exactly the same anyway.

Reply:
If you want to keep it that way, you can replace the two lines by the terser expression:
```suggestion
    vertex_state = fill(unvisited, nv(g))
```